src/ethereum_test_tools/common/types.py

```diff
@@ -1416,7 +1417,7 @@ def payload_body(self) -> List[Any]:
 
         if self.gas_limit is None:
             raise ValueError("gas_limit must be set for all tx types")
-        to = Address(self.to) if self.to is not None else bytes()
+        to = Address(self.to) if self.to else bytes()
```

Comment:
Fixes a bug where if `self.to` was `""`, the value used within `to` would be `b'\x00\x00....\x00'`. This meant the transaction rlp used within `engineNewPayload.executionPayload.transactions[i]` would be invalid, for the case where `tx[i]` had an empty to address (contract creating).

Reply:
Ohh nice! I'll add that - thanks

Reply:
Hmm! It's giving me an error when I add this :(
```
self = typing.Union, args = ('0x0000000000000000000000000000000000000100',), kwds = {}

    def __call__(self, *args, **kwds):
>       raise TypeError(f"Cannot instantiate {self!r}")
E       TypeError: Cannot instantiate typing.Union

../.pyenv/versions/3.10.6/lib/python3.10/typing.py:387: TypeError
```
Can come back to it when we merge to main, as the recent Address changes might help